buildSrc/src/main/java/org/elasticsearch/gradle/util/GradleUtils.java

```diff
@@ -120,4 +132,61 @@ public static void maybeConfigure(TaskContainer tasks, String name, Action<? sup
 
         return (Provider<T>) registration.getService();
     }
+
+    /**
+     * Add a source set and task of the same name that runs tests.
+     *
+     * IDEs are also configured if setup, and the test task is added to check. The new test source
+     * set extends from the normal test source set to allow sharing of utilities.
+     */
+    public static void addTestSourceSet(Project project, String sourceSetName) {
```

Comment:
Can you break up this up into sourceSet creation and task creation methods ? 

I am converting the yamlTests to independent sourceSets and task, and would like to re-use the sourceSet parts here. However, I don't think the task part is re-usabe. 

```
    RestIntegTestTask yamlTestTask = project.getTasks()
            .create("yamlTest", RestIntegTestTask.class, task -> {
                task.dependsOn(project.getTasks().getByName("copyRestApiSpecsTask"));
            });
```
requires a custom `Test.class`, can't use task configuration avoidance (register vs. create) due to the constructor of RestIntegTestTask, needs a dependency. 

I have a WIP for the YAML tests here: https://github.com/elastic/elasticsearch/pull/55833/files#diff-c8d3f49f55d848e6358cf9ebf14b7a79R48

Reply:
I know that we use `RestIntegTestTask`, that exists for a mixture of reasons that I believe are all no longer relevant (eg task ordering which is now solved by testclusters, or common setup which is solved by your plugin instead of inside task creation). Regarding needing a dependency, that can be done by your plugin after the task is created.

Reply:
We could have this method take a `Function<? super SourceSet, ? extends TaskProvider<?>>` factory argument such that the caller defines how the "test task" should be created, and the utility would handle wiring it to `check`, etc.